### PR TITLE
A few Windows improvements

### PR DESCRIPTION
**Pull request by vais**

### CODE REVIEW

File: core/io/shared/tty.rb [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
-      # workaround for not configured environment
     end
-  end
 
-  it "returns false if this stream is not a terminal device (TTY)" do
-    File.open(__FILE__) { |f| f.send(@method) }.should == false
-  end
+    it "returns false if this stream is not a terminal device (TTY)" do
+      File.open(__FILE__) { |f| f.send(@method) }.should == false
+    end
```
Comment (eregon) [inline core/io/shared/tty.rb]: Does this spec and the one below works on windows?
I think only the part within `with_tty` is platform-dependent.

File: core/kernel/system_spec.rb [hunk 2]
```diff
@@ -64,7 +64,7 @@
   platform_is :windows do
     it "runs commands starting with any number of @ using shell" do
       `#{ruby_cmd("p system 'does_not_exist'")} 2>NUL`.chomp.should == "nil"
-      @object.system('@does_not_exist').should == false
+      @object.system('@does_not_exist 2>NUL').should == false
```
Comment (eregon) [inline core/kernel/system_spec.rb]: I wondered a long time how to fix that but it's so simple, congrats!

**Comment by eregon** [comment]
This is great, thanks!

File: core/io/shared/tty.rb (hunk 1 above)
Comment (vais) [inline core/io/shared/tty.rb]: Yep, good catch @eregon, sorry I missed that - just pushed another commit to fix it.

File: core/kernel/system_spec.rb (hunk 2 above)
Comment (vais) [inline core/kernel/system_spec.rb]: :tada:

**Comment by eregon** [comment]
Merged, thanks!

**Pull request merged**